src/core/connection.c

```diff
@@ -1567,8 +1567,12 @@ QuicConnTryClose(
         if (ResultQuicStatus) {
             Connection->CloseStatus = (QUIC_STATUS)ErrorCode;
             Connection->CloseErrorCode = QUIC_ERROR_INTERNAL_ERROR;
-            if (ErrorCode != QUIC_STATUS_CONNECTION_IDLE &&
-                ErrorCode != QUIC_STATUS_CONNECTION_TIMEOUT) {
+            //
+            // Errors 0x1-0xF are quic errors, as are 0x100-0xFF. For future proofing,
+            // consider the whole range of 0x1-0x1FF as a protocol error.
+            //
+            if (ErrorCode > 0 &&
```

Comment:
Notice, we're in the `if (ResultQuicStatus)` block, so `ErrorCode` is not a QUIC protocol error code. Additionally, not all QUIC protocol error codes should be treated as a `QUIC_PERF_COUNTER_CONN_PROTOCOL_ERRORS`. For now, I think 3 through E (inclusive) should be considered protocol errors, as far as this perf counter goes. We should make an inline function `QuicErrorIsProtocolError(...)` right after the error code definitions in frame.h, and put a comment by the function indicating this is for performance counter.

Reply:
Ok. So this should be happening in the else side of that? Adding the function.